op-e2e/interop/interop_benchmark_test.go

```diff
@@ -0,0 +1,214 @@
+	go emitOn(chainA, "Eve")
+	go emitOn(chainB, "Eve")
+	go emitOn(chainA, "Frank")
+	go emitOn(chainB, "Frank")
+	emitParallel.Wait()
+
+	// Get every possible log from both chains, as potential queries
+	// check that the logs are emitted on chain A
+	qA := ethereum.FilterQuery{
+		Addresses: []common.Address{EmitterA},
+	}
+	logsA, err := clientA.FilterLogs(context.Background(), qA)
+	require.NoError(b, err)
+	require.Len(b, logsA, numEmits*6)
+
+	// check that the logs are emitted on chain B
+	qB := ethereum.FilterQuery{
+		Addresses: []common.Address{EmitterB},
+	}
+	logsB, err := clientB.FilterLogs(context.Background(), qB)
+	require.NoError(b, err)
+	require.Len(b, logsB, numEmits*6)
+
+	// helper function to turn a log into an access-list object
+	logToAccessListHashes := func(chainID string, log gethTypes.Log) []common.Hash {
+		client := s2.L2GethClient(chainID, "sequencer")
+		// construct the expected hash of the log's payload
+		// (topics concatenated with data)
+		msgPayload := make([]byte, 0)
+		for _, topic := range log.Topics {
+			msgPayload = append(msgPayload, topic.Bytes()...)
+		}
+		msgPayload = append(msgPayload, log.Data...)
+		msgHash := crypto.Keccak256Hash(msgPayload)
+
+		// get block for the log (for timestamp)
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer cancel()
+		block, err := client.BlockByHash(ctx, log.BlockHash)
+		require.NoError(b, err)
+
+		args := types.ChecksumArgs{
+			BlockNumber: log.BlockNumber,
+			Timestamp:   block.Time(),
+			LogIndex:    uint32(log.Index),
+			ChainID:     eth.ChainIDFromBig(s2.ChainID(chainID)),
+			LogHash:     types.PayloadHashToLogHash(msgHash, log.Address),
+		}
+		accessList := types.EncodeAccessList([]types.Access{args.Access()})
+		return accessList
+	}
+	// get all access lists from both chains
+	allAccessLists := make([][]common.Hash, 0)
+	for _, log := range logsA {
+		allAccessLists = append(allAccessLists, logToAccessListHashes(chainA, log))
+	}
+	for _, log := range logsB {
+		allAccessLists = append(allAccessLists, logToAccessListHashes(chainB, log))
+	}
+
+	super := s2.SupervisorClient()
+
+	executingDescriptor := types.ExecutingDescriptor{Timestamp: uint64(time.Now().Unix() + 1000)}
+	// Run the actual benchmark as a sub-benchmark
+	b.Run("CheckAccessList", func(b *testing.B) {
+		// Reset the timer before the benchmark loop to exclude setup time
+		b.ResetTimer()
+
+		// Run the benchmark loop
+		for i := 0; i < b.N; i++ {
+			// select a random access list
+			randomAccessList := allAccessLists[rand.Intn(len(allAccessLists))]
+			err := super.CheckAccessList(context.Background(), randomAccessList, types.CrossUnsafe, executingDescriptor)
+			require.NoError(b, err)
+		}
+	})
+
+	// Run a benchmark with 10 parallel CheckAccessList calls
+	b.Run("CheckAccessListParallel10", func(b *testing.B) {
+		b.ResetTimer()
+
+		for i := 0; i < b.N; i++ {
+			var wg sync.WaitGroup
+			wg.Add(10)
+
+			for j := 0; j < 10; j++ {
```

Comment:
It would be nice to parametrize the `1, 10, 100` options, to not repeat the benchmark loop code three times.